test cases/d/9 features/meson.build

```diff
@@ -1,10 +1,8 @@
 project('D Features', 'd', default_options : ['debug=false'])
 
-# ONLY FOR BACKWARDS COMPATIBILITY.
-# DO NOT DO THIS IN NEW CODE!
-# USE include_directories() INSTEAD OF BUILDING
-# STRINGS TO PATHS MANUALLY!
-data_dir = join_paths(meson.current_source_dir(), 'data')
+# directory for data
+# This is the correct way to do this.
+data_dir = include_directories('data')
```

Comment:
We can just drop e_plain_bcompat below as well, since it only exists to exercise the manually built path.